migration/identity-migration/src/test/java/io/camunda/migration/identity/handler/sm/RoleMigrationHandlerTest.java

```diff
@@ -0,0 +1,255 @@
+import io.camunda.migration.identity.dto.Permission;
+import io.camunda.migration.identity.dto.Role;
+import io.camunda.security.auth.Authentication;
+import io.camunda.service.AuthorizationServices;
+import io.camunda.service.AuthorizationServices.CreateAuthorizationRequest;
+import io.camunda.service.RoleServices;
+import io.camunda.service.RoleServices.CreateRoleRequest;
+import io.camunda.zeebe.protocol.impl.record.value.authorization.AuthorizationRecord;
+import io.camunda.zeebe.protocol.impl.record.value.authorization.RoleRecord;
+import io.camunda.zeebe.protocol.record.value.AuthorizationOwnerType;
+import io.camunda.zeebe.protocol.record.value.AuthorizationResourceType;
+import io.camunda.zeebe.protocol.record.value.PermissionType;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Answers;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class RoleMigrationHandlerTest {
+
+  private final ManagementIdentityClient managementIdentityClient;
+  private final RoleServices roleServices;
+  private final AuthorizationServices authorizationServices;
+
+  private final RoleMigrationHandler roleMigrationHandler;
+
+  public RoleMigrationHandlerTest(
+      @Mock final ManagementIdentityClient managementIdentityClient,
+      @Mock(answer = Answers.RETURNS_SELF) final RoleServices roleServices,
+      @Mock(answer = Answers.RETURNS_SELF) final AuthorizationServices authorizationServices) {
+    this.managementIdentityClient = managementIdentityClient;
+    this.roleServices = roleServices;
+    this.authorizationServices = authorizationServices;
+    roleMigrationHandler =
+        new RoleMigrationHandler(
+            Authentication.none(), managementIdentityClient, roleServices, authorizationServices);
+  }
+
+  @Test
+  public void shouldMigrateRolesAndAuthorizations() {
+    // given
+    when(managementIdentityClient.fetchRoles())
+        .thenReturn(
+            List.of(
+                new Role("Role 1", "Description for Role 1"),
+                new Role(
+                    "Role@Name#With$Special%Chars", "Description for Role with special chars")));
+    when(roleServices.createRole(any()))
+        .thenReturn(CompletableFuture.completedFuture(new RoleRecord()));
+    when(managementIdentityClient.fetchPermissions(any()))
+        .thenReturn(
+            List.of(
+                new Permission("read", "camunda-identity-resource-server"),
+                new Permission("read:users", "camunda-identity-resource-server"),
+                new Permission("write", "camunda-identity-resource-server"),
+                new Permission("read:*", "operate-api"),
+                new Permission("write:*", "operate-api")))
+        .thenReturn(
```

Comment:
❓ to confirm I got it right, semantics here are that the second invocation returns the content of the second `thenReturn` ?